rs/tests/cross_chain/ic_xc_cketh_test.rs

```diff
@@ -0,0 +1,984 @@
+use ic_system_test_driver::driver::test_env_api::SubnetSnapshot;
+use ic_system_test_driver::driver::universal_vm::DeployedUniversalVm;
+use ic_system_test_driver::util::MessageCanister;
+use ic_system_test_driver::{
+    driver::{
+        group::SystemTestGroup,
+        ic::{InternetComputer, Subnet},
+        test_env::TestEnv,
+        test_env_api::{
+            get_dependency_path, HasPublicApiUrl, HasTopologySnapshot, IcNodeContainer,
+            NnsCustomizations, SshSession,
+        },
+        universal_vm::{UniversalVm, UniversalVms},
+    },
+    systest,
+    util::{block_on, runtime_from_url},
+};
+use ic_types::Height;
+use icrc_ledger_types::icrc1::account::Account;
+use reqwest::Client;
+use slog::{info, Logger};
+use std::env;
+use std::future::Future;
+use std::time::Duration;
+
+const FOUNDRY_VM_NAME: &str = "foundry";
+const DOCKER_NETWORK_NAME: &str = "ethereum";
+const FOUNDRY_PORT: u16 = 8545;
+const ENCODED_PRINCIPAL: &str =
+    "0x1d9facb184cbe453de4841b6b9d9cc95bfc065344e485789b550544529020000";
+
+fn main() -> Result<()> {
+    SystemTestGroup::new()
+        .with_setup(setup_with_system_and_application_subnets)
+        .add_test(systest!(ic_xc_cketh_test))
+        .execute_from_args()?;
+    Ok(())
+}
+
+fn setup_with_system_and_application_subnets(env: TestEnv) {
+    setup_anvil(&env);
+    InternetComputer::new()
+        .add_subnet(Subnet::new(SubnetType::System).add_nodes(1))
+        .add_subnet(
+            Subnet::new(SubnetType::Application)
+                .add_nodes(1)
+                .with_dkg_interval_length(Height::from(10)),
+        )
+        .setup_and_start(&env)
+        .expect("Failed to setup IC under test");
+    install_nns_with_customizations_and_check_progress(
+        env.topology_snapshot(),
+        NnsCustomizations::default(),
+    );
+
+    env.topology_snapshot()
+        .subnets()
+        .for_each(|subnet| subnet.await_all_nodes_healthy().unwrap());
+}
+
+fn setup_anvil(env: &TestEnv) {
+    UniversalVm::new(String::from(FOUNDRY_VM_NAME))
+        .with_config_img(get_dependency_path(
+            env::var("CKETH_UVM_CONFIG_PATH").expect("CKETH_UVM_CONFIG_PATH not set"),
+        ))
+        .enable_ipv4() //forge needs to download the version of the solidity compiler indicated in the smart contracts that are being deployed
+        .start(env)
+        .expect("failed to setup universal VM");
+
+    let deployed_universal_vm = env.get_deployed_universal_vm(FOUNDRY_VM_NAME).unwrap();
+
+    println!(
+        "{}",
+        deployed_universal_vm
+            .block_on_bash_script(&format!(
+                r#"
+docker load -i /config/foundry.tar
+docker network create {DOCKER_NETWORK_NAME}
+docker run --net {DOCKER_NETWORK_NAME} --detach --rm --name anvil -p {FOUNDRY_PORT}:{FOUNDRY_PORT} foundry:latest "anvil --host 0.0.0.0"
+docker logs anvil
+"#
+            ))
+            .unwrap()
+    );
+}
+
+fn ic_xc_cketh_test(env: TestEnv) {
```

Comment:
It would be nice to have a few log messages in this function (`info!(logger, "bla bla bla ...");`) to better pinpoint where the test is failing in case it fails.

Reply:
There are a few logs statement inside, in particular in the `test_` methods but I can always add a bit more.